### PR TITLE
Adding Video Functionality

### PR DESCRIPTION
**Pull request by superchilled**

### CODE REVIEW

File: lib/vonage/video.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    self.host = :video_host
+
+    # Generate a new session.
+    #
+    # @example
+    #   session = client.video.create_session({
+    #     archive_mode: 'always',
+    #     location: '10.1.200.30',
+    #     p2p_preference: 'disabled'
```
Comment (dragonmantank) [inline lib/vonage/video.rb]: We don't actually expose this through the SDKs, but instead expose `mediaMode`, which is either `routed` or `relayed`. This then swaps `p2p_preference` to `disabled` or `enabled`, respectively, under the hood.

File: lib/vonage/video.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    #
+    def create_session(**params)
+      request_params = params.clone
+      p2p_preference = request_params.delete(:p2p_preference)
+      request_params['p2p.preference'] = p2p_preference if p2p_preference
+
+      response = request('/session/create', params: camelcase(request_params), type: Post)
+
+      params.keys.each {|key| response.entity[key] = params[key]}
+      response
```
Comment (dragonmantank) [inline lib/vonage/video.rb]: This is another oddity of creating a session - we don't return the full response in the SDKs. We only return:

* `sessionId`
* `archiveMode`
* `mediaMode`
* `location`

File: lib/vonage/video.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    end
+
+    def generate_client_token(session_id:, application_id: @config.application_id, private_key: @config.private_key, scope: 'session.connect', role: 'publisher', **params)
```
Comment (dragonmantank) [inline lib/vonage/video.rb]: Why are we asking the dev to pass in the private key? Shouldn't this already be a part of their auth that we can pull this out of?

If users want to make just a general JWT, they should sue the JWT library directly.

File: lib/vonage/video/archives.rb [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    #
+    # @see TODO: add docs link
+    #
+    def info(application_id: @config.application_id, archive_id:)
+      # TODO: raise error if application_id is nil
+
+      request('/v2/project/' + application_id + '/archive/' + archive_id)
+    end
+
+    # Create a new archive.
```
Comment (dragonmantank) [inline lib/vonage/video/archives.rb:52]: We need to make sure we support the newer multiArchive tag

**Review by dragonmantank — CHANGES_REQUESTED** [review]
Couple of changes to creating sessions and archives

File: lib/vonage/video.rb (hunk 1 above)
Comment (superchilled) [reply lib/vonage/video.rb]: I've updated the logic for this and added some tests to cover the new logic.

File: lib/vonage/video.rb (hunk 2 above)
Comment (superchilled) [reply lib/vonage/video.rb]: [As above](https://github.com/Vonage/vonage-ruby-sdk/pull/249#discussion_r1043298268)

File: lib/vonage/video.rb (hunk 3 above)
Comment (superchilled) [reply lib/vonage/video.rb]: > Shouldn't this already be a part of their auth that we can pull this out of?

Yeah, it should be set in the `Config` object, and I do set the default value for the param as `@config.private_key`, though I guess the question is whether we need to even expose this param at all (same goes for `application_id`). I think I was just following the same logic that already exists in the [JWT.generate](https://github.com/Vonage/vonage-ruby-sdk/blob/5dd438f058cbffa7bb92caccf0ed703b40a011ea/lib/vonage/jwt.rb#L36) method signature, though I'm now wondering why it's even exposed there.

Will update the method sig.

File: lib/vonage/video/archives.rb (hunk 4 above)
Comment (superchilled) [reply lib/vonage/video/archives.rb:52]: Updated the code comments.

**Review by dragonmantank — APPROVED** [review]
LGTM

**Comment by superchilled** [comment]
Closing the PR as not merging into `dev` yet, but instead releasing a `beta` version of the library.

**Pull request closed without merging**